Image counter-clockwise rotation

nontas opened the pull request:
Till now, the `Image` class had the `rotate_ccw_about_centre()` method that basically uses `rotate_ccw_about_centre()`. However, the `Image` that is returned from this method has the exact same shape as the original one. This means that naturally some regions of the original image get cropped, depending on the rotation angle.

This PR adds another method, called `rotate_ccw()`, which does the rotation but without any cropping. The methods creates the correct `Transform` so that the output image has the proper shape in order to contain all the information from the original image. This is the `Transform` that involves:

1. Translation to the origin (thus by ``-self.centre()``)
2. Counter-clockwise rotation
3. Apply steps 1 and 2 to the image's bounding box to get `rotated_bbox`
4. Translation by ``-rotated_bbox.bounds()[0]``

#### Example
```
%matplotlib inline
import menpo.io as mio
import matplotlib.pyplot as plt

# Load image
image = mio.import_builtin_asset.lenna_png()
image.crop_to_landmarks_proportion_inplace(0.2)

# Options
theta = 60
degrees=True

# Rotation
rot_image_1 = image.rotate_ccw(theta, degrees)

# Rotation about centre
rot_image_2 = image.rotate_ccw_about_centre(theta, degrees)

# View
plt.subplot(121)
rot_image_1.view_landmarks()

plt.subplot(122)
rot_image_2.view_landmarks()
```

patricksnape commented:
What if this was a kwarg instead of a new method?

nontas commented:
OK I'll do that

nontas commented:
Done. I defined a single method, called `rotate_ccw()` (I changed the name from `rotate_ccw_about_centre()` because this is not true any more). The method has a `retain_shape` bool argument that controls the shape of the output image.

patricksnape commented on menpo/image/base.py:
```diff
@@ -1788,19 +1788,35 @@ def zoom(self, scale, cval=0.0):
+                   rotate_landmarks=True):
         r"""
-        Return a rotation of this image clockwise about its centre.
+        Return a counter-clockwise rotation of this image.
```
It's probably worth noting that this **is** around the image centre. Because technically a rotation would be around the origin and would rotate the image out of the frame.

patricksnape commented:
I'm cool with this - I think it's good behaviour. What does @jabooth think about the method name change? Should would be deprecating the old method?

nontas commented:
Actually this is not ready yet! I think my implementation is wrong and I'm not sure yet what's the mistake... It's easier to discuss it in person...

nontas commented:
After the discussion with @patricksnape and @jabooth , the following are done:
* Added `order` and `round` arguments similar to the `rescale` method.
* Kept the `rotate_ccw_about_centre` name.
* Removed the `warp_landmarks` from all methods apart from the warping ones.
* Set `warp_landmarks` default value to ``True`` for `warp_to_shape` and `warp_to_mask`.
* Added tests.

Let me know if you guys are happy

patricksnape commented:
I think it is still just cutting off the corners... But as we discussed, I'm not sure this matters that much? I'm happy to get this in as a convenience as see if anyone has a real problem with its current behaviour.

patricksnape commented:
I think it is still just cutting off the corners... But as we discussed, I'm not sure this matters that much? I'm happy to get this in as a convenience as see if anyone has a real problem with its current behaviour. +1

nontas commented:
As discussed with @patricksnape , the images do not get cropped in case `retain_shape=False`. It's just that sometimes, an extra row is added on their bottom part due to the transform. This is a good example:
```
from menpo.image import Image
import numpy as np

image = Image(np.array([[ 1., 2., 3., 4.], [5., 6., 7., 8.], [9., 10., 11., 12.], 
                        [13., 14., 15., 16.]]))
rotated_image = image.rotate_ccw_about_centre(45, order=0, retain_shape=False)
print image.pixels
print rotated_image.pixels
```
which prints
```
[[[  1.   2.   3.   4.]
  [  5.   6.   7.   8.]
  [  9.  10.  11.  12.]
  [ 13.  14.  15.  16.]]]
[[[  0.   0.   4.   0.   0.]
  [  0.   3.   7.   8.   0.]
  [  1.   6.   7.  11.  16.]
  [  0.   5.  10.  15.  15.]
  [  0.   0.  13.  14.   0.]]]
```
However, by using a 5x5 array as
```
image = Image(np.array([[ 1.,  2.,  3.,  4.,  5.], [ 6.,  7.,  8.,  9., 10.], 
                        [11., 12., 13., 14., 15.], [16., 17., 18., 19., 20.], 
                        [21., 22., 23., 24., 25.]]))
```
everything looks fine:
```
[[[  1.   2.   3.   4.   5.]
  [  6.   7.   8.   9.  10.]
  [ 11.  12.  13.  14.  15.]
  [ 16.  17.  18.  19.  20.]
  [ 21.  22.  23.  24.  25.]]]
[[[  0.   0.   0.   5.   0.   0.   0.]
  [  0.   0.   4.   9.  15.   0.   0.]
  [  0.   2.   8.  14.  14.  20.   0.]
  [  1.   7.  12.  13.  19.  24.  25.]
  [  0.  11.  12.  17.  23.  24.   0.]
  [  0.   0.  16.  22.  22.   0.   0.]
  [  0.   0.   0.  21.   0.   0.   0.]]]
```

I believe that the method is good to get in, since this behaviour is not really that important.
